bundles/config/org.eclipse.smarthome.config.discovery.usbserial.linuxsysfs/src/main/java/org/eclipse/smarthome/config/discovery/usbserial/linuxsysfs/internal/SysfsUsbSerialScanner.java

```diff
@@ -0,0 +1,269 @@
+ * <p/>
+ * If the device providing the serial port is a USB device, information about the device (vendor ID, product ID, etc.)
+ * is collected from the sysfs and returned together with the name of the serial port in form of a
+ * {@link UsbSerialDeviceInformation}.
+ *
+ * @author Henning Sudbrock - initial contribution
+ */
+@Component(configurationPid = "discovery.usbserial.linuxsysfs.usbserialscanner")
+@NonNullByDefault
+public class SysfsUsbSerialScanner implements UsbSerialScanner {
+
+    private final Logger logger = LoggerFactory.getLogger(SysfsUsbSerialScanner.class);
+
+    public static final String SYSFS_TTY_DEVICES_DIRECTORY_ATTRIBUTE = "sysfsTtyDevicesPath";
+    public static final String DEV_DIRECTORY_ATTRIBUTE = "devPath";
+
+    private static final String SYSFS_TTY_DEVICES_DIRECTORY_DEFAULT = "/sys/class/tty";
+    private static final String DEV_DIRECTORY_DEFAULT = "/dev";
+
+    private String sysfsTtyDevicesDirectory = SYSFS_TTY_DEVICES_DIRECTORY_DEFAULT;
+    private String devDirectory = DEV_DIRECTORY_DEFAULT;
+
+    private static final String SYSFS_FILENAME_USB_VENDOR_ID = "idVendor";
+    private static final String SYSFS_FILENAME_USB_PRODUCT_ID = "idProduct";
+    private static final String SYSFS_FILENAME_USB_SERIAL_NUMBER = "serial";
+    private static final String SYSFS_FILENAME_USB_MANUFACTURER = "manufacturer";
+    private static final String SYSFS_FILENAME_USB_PRODUCT = "product";
+
+    /**
+     * In the sysfs, directories for USB interfaces have the following format (cf., e.g.,
+     * http://www.linux-usb.org/FAQ.html#i6), where there can be one or more USB port numbers separated by dots.
+     *
+     * <pre>
+     * {@code <#bus>-<#port>.<#port>:<#config>:<#interface>}
+     * </pre>
+     *
+     * Example: {@code 3-1.3:1.0}
+     * <p/>
+     * This format is captured by this {@link Pattern}.
+     */
+    private static final Pattern SYSFS_USB_INTERFACE_DIRECTORY_PATTERN = Pattern
+            .compile("\\d+-(\\d+\\.?)*\\d+:\\d+\\.\\d+");
+
+    @Activate
+    protected void activate(Map<String, Object> config) {
+        extractConfiguration(config);
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        extractConfiguration(config);
+    }
+
+    @Override
+    public Set<UsbSerialDeviceInformation> scan() throws IOException {
+        Set<UsbSerialDeviceInformation> result = Sets.newHashSet();
+
+        for (SerialPortInfo serialPortInfo : getSerialPortInfos()) {
+            try {
+                UsbSerialDeviceInformation usbSerialDeviceInfo = tryGetUsbSerialDeviceInformation(serialPortInfo);
+                if (usbSerialDeviceInfo != null) {
+                    result.add(usbSerialDeviceInfo);
+                }
+            } catch (IOException e) {
+                logger.warn(
+                        "Could not extract usb device information for serial port {} due to an IOException; message: {}",
```

Comment:
"Could not extract **USB** device information for serial port {}: ~~due to an IOException; message:~~ {}"